test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -107,6 +114,65 @@ public void CheckStateForUnstructuredLogWithStringInterpolation()
             Assert.Equal(message.ToString(), state.ToString());
         }
 
+        [Fact]
+        public void StateCanBeSetByProcessor()
+        {
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(Field.State));
+                    options.AddInMemoryExporter(this.exportedItems);
+                }));
+
+            var logger = loggerFactory.CreateLogger<LogRecordTest>();
+            {
+                logger.LogInformation($"This should be replaced {0.99}.");
+
+                var state = this.exportedItems[0].State as IReadOnlyList<KeyValuePair<string, object>>;
+                Assert.Equal("newStateKey", state[0].Key.ToString());
+                Assert.Equal("newStateValue", state[0].Value.ToString());
+            }
+        }
+
+        [Fact]
+        public void StateValuesCanBeSetByProcessor()
+        {
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(Field.StateValues));
+                    options.AddInMemoryExporter(this.exportedItems);
+                }));
+
+            var logger = loggerFactory.CreateLogger<LogRecordTest>();
+            logger.Log(
+                LogLevel.Information,
+                0,
+                new List<KeyValuePair<string, object>> { new KeyValuePair<string, object>("OldKey", "OldValue") },
+                null,
+                (s, e) => "OpenTelemetry!");
+
+            var stateValue = this.exportedItems[0];
+            Assert.Equal(new KeyValuePair<string, object>("newStateValueKey", "newStateValueValue"), stateValue.StateValues[0]);
+        }
+
+        [Fact]
+        public void FormattedMessageCanBeSetByProcessor()
+        {
+            using var loggerFactory = LoggerFactory.Create(builder => builder
+                .AddOpenTelemetry(options =>
+                {
+                    options.AddProcessor(new RedactionProcessor(Field.FormattedMessage));
```

Comment:
similar comment here - if the goal is FormattedMessage test, force SDK to populate it - options.IncludeFormattedMessage = true